examples/inertia_relief/inertia_relief_example.cpp

```diff
@@ -0,0 +1,542 @@
+  writer.write(0, 0.0, objective_states);
+  auto non_const_states = getFieldPointers(states, params);
+  InertialReliefProblem problem({non_const_states[SHAPE_DISP], non_const_states[DISP], non_const_states[DENSITY]},
+                                non_const_states, solid_mechanics_residual, constraints);
+  int dimx = problem.GetDimx();
+  int dimy = problem.GetDimy();
+
+  mfem::Vector x0(dimx);
+  x0 = 0.0;
+  mfem::Vector y0(dimy);
+  y0 = 0.0;
+  mfem::Vector xf(dimx);
+  xf = 0.0;
+  mfem::Vector yf(dimy);
+  yf = 0.0;
+
+  HomotopySolver solver(&problem);
+  double tol = 1.e-6;
+  int maxIter = 100;
+  solver.SetTol(tol);
+  solver.SetMaxIter(maxIter);
+
+  solver.Mult(x0, y0, xf, yf);
+  bool converged = solver.GetConverged();
+  if (myid == 0) {
+    if (converged) {
+      std::cout << "converged!\n";
+    } else {
+      std::cout << "homotopy solver did not converge\n";
+    }
+  }
+  writer.write(1, 1.0, objective_states);
+}
+
+InertialReliefProblem::InertialReliefProblem(std::vector<serac::FiniteElementState*> obj_states_,
+                                             std::vector<serac::FiniteElementState*> all_states_,
+                                             std::shared_ptr<serac::Residual> residual_,
+                                             std::vector<std::shared_ptr<serac::ScalarObjective>> constraints_)
+    : GeneralNLMCProblem(),
+      dFdx(nullptr),
+      dFdy(nullptr),
+      dQdx(nullptr),
+      dQdy(nullptr),
+      uOffsets(nullptr),
+      cOffsets(nullptr)
+{
+  residual = residual_;
+  constraints.resize(constraints_.size());
+  std::copy(constraints_.begin(), constraints_.end(), constraints.begin());
+
+  all_states.resize(all_states_.size());
+  std::copy(all_states_.begin(), all_states_.end(), all_states.begin());
+
+  obj_states.resize(obj_states_.size());
+  std::copy(obj_states_.begin(), obj_states_.end(), obj_states.begin());
+
+  int numConstraints = static_cast<int>(constraints.size());
+  uOffsets = new HYPRE_BigInt[2];
+  cOffsets = new HYPRE_BigInt[2];
+  for (int i = 0; i < 2; i++) {
+    uOffsets[i] = all_states[FIELD::DISP]->space().GetTrueDofOffsets()[i];
+  }
+
+  int myid = mfem::Mpi::WorldRank();
+  if (myid == 0) {
+    cOffsets[0] = 0;
+  } else {
+    cOffsets[0] = numConstraints;
+  }
+  cOffsets[1] = numConstraints;
+
+  dimu = uOffsets[1] - uOffsets[0];
+  dimc = cOffsets[1] - cOffsets[0];
+  y_partition.SetSize(3);
+  y_partition[0] = 0;
+  y_partition[1] = dimu;
+  y_partition[2] = dimc;
+  y_partition.PartialSum();
+
+  {
+    HYPRE_BigInt* dofOffsets = new HYPRE_BigInt[2];
+    HYPRE_BigInt* complementarityOffsets = new HYPRE_BigInt[2];
+    for (int i = 0; i < 2; i++) {
+      dofOffsets[i] = uOffsets[i] + cOffsets[i];
+    }
+    complementarityOffsets[0] = 0;
+    complementarityOffsets[1] = 0;
+    Init(complementarityOffsets, dofOffsets);
+    delete[] dofOffsets;
+    delete[] complementarityOffsets;
```

Comment:
```suggestion
    HYPRE_BigInt dofOffsets[2];
    HYPRE_BigInt complementarityOffsets[2];
    for (int i = 0; i < 2; i++) {
      dofOffsets[i] = uOffsets[i] + cOffsets[i];
    }
    complementarityOffsets[0] = 0;
    complementarityOffsets[1] = 0;
    Init(&complementarityOffsets, &dofOffsets);
```
You should avoid new/delete when possible. It will be destroyed when it falls out of the scope you created.